RiotSwiftUI/Modules/Template/SimpleUserProfileExample/Coordinator/TemplateUserProfileCoordinator.swift

```diff
@@ -0,0 +1,64 @@
+/*
+ Copyright 2021 New Vector Ltd
+ 
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+ 
+ http://www.apache.org/licenses/LICENSE-2.0
+ 
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import Foundation
+import UIKit
+import SwiftUI
+
+final class TemplateUserProfileCoordinator: Coordinator {
+    
+    // MARK: - Properties
+    
+    // MARK: Private
+    
+    private let session: MXSession
+    private let templateUserProfileViewController: UIViewController
+    private var templateUserProfileViewModel: TemplateUserProfileViewModelProtocol
+    
+    // MARK: Public
+
+    // Must be used only internally
+    var childCoordinators: [Coordinator] = []
+    var completion: (() -> Void)?
+    
+    // MARK: - Setup
+    
+    @available(iOS 14.0, *)
+    init(session: MXSession) {
```

Comment:
We are now using an input parameters class for Coodinators

Reply:
I saw the PR for that but couldn't see the discussion you referenced before. Any details of that just for context of those changes?

Reply:
Yes we are now using a `struct` as input parameter
```
struct TemplateUserProfileCoordinatorParamers {
 let session: MXSession
}
```
To make init refactoring easier and handle default values and convenience init aside.